pkg/trace/config/config.go

```diff
@@ -213,13 +210,22 @@ func (c *AgentConfig) acquireHostname() error {
 	cmd.Stdout = &out
 	err := cmd.Run()
 	c.Hostname = strings.TrimSpace(out.String())
-	if err != nil || c.Hostname == "" {
-		c.Hostname, err = fallbackHostnameFunc()
-	}
-	if c.Hostname == "" {
-		err = ErrMissingHostname
+	if hostnameIsEmpty := features.Has("disable_empty_hostname") && c.Hostname == ""; err != nil || hostnameIsEmpty {
```

Comment:
naming is a bit weird. hostnameIsEmpty should be `c.Hostname == ""` based on the name

Reply:
Same here, I agree, but couldn't find a better name. What about `hostnameDisallowed`?